server/server.go

```diff
@@ -1781,6 +1783,13 @@ func (s *Server) campaignLeader() {
 				log.Info("etcd leader changed, resigns pd leadership", zap.String("old-pd-leader-name", s.Name()))
 				return
 			}
+			// check healthy status of etcd leader.
+			if s.member.GetLeadership().GetUnHealthyTimesNum() > unhealthyLeaderLeaseTimes {
```

Comment:
What about doing this check at the beginning of the tick to make sure we could resign the etcd leader as soon as possible?

Reply:
I think 50ms maybe can be ignored